egs/librispeech/tts1/run.sh

```diff
@@ -0,0 +1,328 @@
+            --mfcc-config conf/mfcc.conf \
+            --nj ${nj} --cmd "$train_cmd" \
+            data/${name}_mfcc exp/make_mfcc $mfccdir
+        utils/fix_data_dir.sh data/${name}_mfcc
+        sid/compute_vad_decision.sh --nj ${nj} --cmd "$train_cmd" \
+            data/${name}_mfcc exp/make_vad ${vaddir}
+        utils/fix_data_dir.sh data/${name}_mfcc
+    done
+    # Check pretrained model existence
+    nnet_dir=exp/xvector_nnet_1a
+    if [ ! -e $nnet_dir ];then
+        echo "X-vector model does not exist. Download pre-trained model."
+        wget http://kaldi-asr.org/models/8/0008_sitw_v2_1a.tar.gz
+        tar xvf 0008_sitw_v2_1a.tar.gz
+        mv 0008_sitw_v2_1a/exp/xvector_nnet_1a exp
+        rm -rf 0008_sitw_v2_1a.tar.gz 0008_sitw_v2_1a
+    fi
+    # Extract x-vector
+    for name in ${train_set} ${train_dev} ${eval_set}; do
+        sid/nnet3/xvector/extract_xvectors.sh --cmd "$train_cmd --mem 4G" --nj ${nj} \
+            $nnet_dir data/${name}_mfcc \
+            $nnet_dir/xvectors_${name}
+    done
+    # Update json
+    for name in ${train_set} ${train_dev} ${eval_set}; do
+        local/update_json.sh ${dumpdir}/${name}/data.json ${nnet_dir}/xvectors_${name}/xvector.scp
+    done
+fi
+
+
+if [ -z ${tag} ];then
+    expdir=exp/${train_set}_taco2_enc${embed_dim}
+    if [ ${econv_layers} -gt 0 ];then
+        expdir=${expdir}-${econv_layers}x${econv_filts}x${econv_chans}
+    fi
+    expdir=${expdir}-${elayers}x${eunits}_dec${dlayers}x${dunits}
+    if [ ${prenet_layers} -gt 0 ];then
+        expdir=${expdir}_pre${prenet_layers}x${prenet_units}
+    fi
+    if [ ${postnet_layers} -gt 0 ];then
+        expdir=${expdir}_post${postnet_layers}x${postnet_filts}x${postnet_chans}
+    fi
+    expdir=${expdir}_att${adim}-${aconv_filts}x${aconv_chans}
+    if ${cumulate_att_w};then
+        expdir=${expdir}_cm
+    fi
+    if ${use_batch_norm};then
+        expdir=${expdir}_bn
+    fi
+    if ${use_residual};then
+        expdir=${expdir}_rs
+    fi
+    if ${use_concate};then
+        expdir=${expdir}_cc
+    fi
+    if ${use_masking};then
+        expdir=${expdir}_msk_pw${bce_pos_weight}
+    fi
+    expdir=${expdir}_do${dropout}_zo${zoneout}_lr${lr}_ep${eps}_wd${weight_decay}_bs$((batchsize*ngpu))
+    if [ ! -z ${batch_sort_key} ];then
+        expdir=${expdir}_sort_by_${batch_sort_key}_mli${maxlen_in}_mlo${maxlen_out}
+    fi
+    expdir=${expdir}_sd${seed}
+else
+    expdir=exp/${train_set}_${tag}
+fi
+if [ ${stage} -le 4 ];then
+    echo "stage 4: Text-to-speech model training"
+    tr_json=${feat_tr_dir}/data.json
+    dt_json=${feat_dt_dir}/data.json
+    ${cuda_cmd} --gpu ${ngpu} ${expdir}/train.log \
+        tts_train.py \
+           --ngpu ${ngpu} \
+           --outdir ${expdir}/results \
+           --verbose ${verbose} \
+           --seed ${seed} \
+           --resume ${resume} \
+           --train-json ${tr_json} \
+           --valid-json ${dt_json} \
+           --embed_dim ${embed_dim} \
```

Comment:
another way is to add the embedding dimension information (shape) in the JSON file. It is redundant, but it is more consistent with the normal feature.